### PR TITLE
Move frozen indices to x-pack module

### PR DESCRIPTION
**Pull request by ywelsch**
Will facilitate future development of this feature, e.g. backporting of https://github.com/elastic/elasticsearch/pull/44286

### CODE REVIEW

File: x-pack/plugin/frozen-indices/src/main/java/org/elasticsearch/xpack/frozen/FrozenIndices.java [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+        return actions;
+    }
+
+    @Override
+    public List<RestHandler> getRestHandlers(Settings settings, RestController restController, ClusterSettings clusterSettings,
+                                             IndexScopedSettings indexScopedSettings, SettingsFilter settingsFilter,
+                                             IndexNameExpressionResolver indexNameExpressionResolver,
+                                             Supplier<DiscoveryNodes> nodesInCluster) {
+        List<RestHandler> handlers = new ArrayList<>();
+        handlers.add(new RestFreezeIndexAction(settings, restController));
```
Comment (dnhatn) [inline x-pack/plugin/frozen-indices/src/main/java/org/elasticsearch/xpack/frozen/FrozenIndices.java]: perhaps use `singletonList`?

File: x-pack/plugin/frozen-indices/src/main/java/org/elasticsearch/xpack/core/frozen/rest/action/RestFreezeIndexAction.java [hunk 2]
```diff
@@ -3,19 +3,19 @@
  * or more contributor license agreements. Licensed under the Elastic License;
  * you may not use this file except in compliance with the Elastic License.
  */
-package org.elasticsearch.xpack.core.rest.action;
+package org.elasticsearch.xpack.core.frozen.rest.action;
```
Comment (dnhatn) [inline x-pack/plugin/frozen-indices/src/main/java/org/elasticsearch/xpack/core/frozen/rest/action/RestFreezeIndexAction.java]: I am not sure if this package needs to be under "core".

**Review by dnhatn — APPROVED** [review]
LGTM. I left two optional comments.

File: x-pack/plugin/frozen-indices/src/main/java/org/elasticsearch/xpack/frozen/FrozenIndices.java (hunk 1 above)
Comment (ywelsch) [reply x-pack/plugin/frozen-indices/src/main/java/org/elasticsearch/xpack/frozen/FrozenIndices.java]: done in a8d8356

**Pull request merged**